src/Aspire.Dashboard/Model/DashboardClient.cs

```diff
@@ -300,11 +301,11 @@ ResourceViewModelSubscription IDashboardClient.SubscribeResources()
                 InitialState: _resourceByName.Values.ToImmutableArray(),
                 Subscription: StreamUpdates());
 
-            async IAsyncEnumerable<ResourceViewModelChange> StreamUpdates()
+            async IAsyncEnumerable<ResourceViewModelChange> StreamUpdates([EnumeratorCancellation] CancellationToken cancellationToken = default)
```

Comment:
How would this ever be anything other than `CancellationToken.None`, based on the call site above not actually passing an argument?

Reply:
Because of [[EnumeratorCancellation]](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.enumeratorcancellationattribute?view=net-8.0).

You can see it's not an empty CT in the debugger:
![image](https://github.com/dotnet/aspire/assets/303201/7cdc9e3a-8344-4e28-8e55-7a58222481ac)

Reply:
Yes, the attribute is info for the compiler when creating the real code.

I think a token passed on 302 wouldn't do anything. An analyzer probably tells you to not ever try.

Reply:
Passing a token directly doesn't give a warning, and actually works as you might expect. In fact, you can pass one token directly and another via `WithCancellation` and cancelling either of them will cancel the enumeration. They're combined into a single token internally.